### ISSUE TITLE
Weight decay in adam optimizer

### ISSUE THREAD

**Issue by JoaoLages** [tensorflow/tensorflow#33110]
- TensorFlow version: 1.14 / 2.*
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
Could we please get weight decay (L2 penalty) in Adam optimizer? PyTorch has it and it is as easy as adding a `weight_decay` float argument when initializing the optimizer.
Rectified adam would be great too! 

**Who will benefit with this feature?**
Everyone that want to use L2 weight decay.

**Comment by gowthamkpr** [comment]
@JoaoLages Is this a feature request? Can you please fill in the issue template thats given [here](https://github.com/tensorflow/tensorflow/issues/new/choose) which will help us expedite the process. Thanks!

**Comment by JoaoLages** [comment]
@gowthamkpr I've updated the issue description, idk if this is what you were looking for

**Comment by gowthamkpr** [comment]
@JoaoLages Thanks for the update.

**Comment by tanzhenyu** [comment]
Please use this [optimizer](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/weight_decay_optimizers.py)

**Comment by JoaoLages** [comment]
@tanzhenyu can you provide an example of how to use it?

**Comment by bonlime** [comment]
@JoaoLages it should be trivial. just like any other optimizer
